### PR TITLE
support userscramcredentials apis

### PR DESCRIPTION
**Pull request by petedannemann**
Support userscramcredentials APIs

[DescribeUserScramCredentials API Docs](https://kafka.apache.org/protocol#The_Messages_DescribeUserScramCredentials)

[AlterUserScramCredentials API Docs](https://kafka.apache.org/protocol#The_Messages_AlterUserScramCredentials)

I based this PR on similar work done here https://github.com/segmentio/kafka-go/pull/1119

### CODE REVIEW

File: describeuserscramcredentials.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	Throttle time.Duration
+
+	// Top level error that occurred while attempting to describe
+	// the user scram credentials.
+	//
+	// The errors contain the kafka error code. Programs may use the standard
+	// errors.Is function to test the error against kafka error codes.
+	Error error
+
+	// List of describeed user scram credentials.
```
Comment (hhahn-tw) [inline describeuserscramcredentials.go:39]: > describeed

described

File: alteruserscramcredentials_test.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+import (
+	"context"
+	"testing"
+
+	ktesting "github.com/segmentio/kafka-go/testing"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAlterUserScramCredentials(t *testing.T) {
```
Comment (hhahn-tw) [inline alteruserscramcredentials_test.go:11]: Would it be of value to test deletions as well?

Comment (petedannemann) [reply alteruserscramcredentials_test.go:11]: Good idea, just added them

**Review by hhahn-tw — APPROVED** [review]
Looks good!

File: alteruserscramcredentials_test.go [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+	if len(deleteRes.Results) != 1 {
+		t.Fatalf("expected 1 deleteResult; got %d", len(deleteRes.Results))
+	}
+
+	if deleteRes.Results[0].User != "alice" {
+		t.Fatalf("expected deleteResult with user alice, got %s", deleteRes.Results[0].User)
+	}
+
+	describeDeletionRes, err := client.DescribeUserScramCredentials(context.Background(), &DescribeUserScramCredentialsRequest{
```
Comment (petedannemann) [inline alteruserscramcredentials_test.go:109]: Move this to describeuserscramcredentials_test.go

File: alteruserscramcredentials_test.go [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+		return
+	}
+
+	client, shutdown := newLocalClient()
+	defer shutdown()
+
+	createRes, err := client.AlterUserScramCredentials(context.Background(), &AlterUserScramCredentialsRequest{
+		Upsertions: []UserScramCredentialsUpsertion{
+			{
+				Name:           "alice",
```
Comment (petedannemann) [inline alteruserscramcredentials_test.go:24]: Generate a unique principal name to avoid test collisions

File: alteruserscramcredentials.go [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+type UserScramCredentialsDeletion struct {
+	Name      string
+	Mechanism ScramMechanism
+}
+
+type UserScramCredentialsUpsertion struct {
+	Name           string
+	Mechanism      ScramMechanism
+	Iterations     int32
```
Comment (rhansen2) [inline alteruserscramcredentials.go:41]: I think we're a bit inconsistent but I believe we try to lean towards int in the Client types in the case where an int32 is needed.

File: alteruserscramcredentials.go [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+type AlterUserScramCredentialsResponse struct {
+	// The amount of time that the broker throttled the request.
+	Throttle time.Duration
+
+	// List of errors that occurred while attempting to alter
+	// the user scram credentials.
+	//
+	// The errors contain the kafka error code. Programs may use the standard
+	// errors.Is function to test the error against kafka error codes.
+	Errors []error
```
Comment (rhansen2) [inline alteruserscramcredentials.go:57]: I think we tend to keep errors paired with their individual response instead of separating them here.

**Review by rhansen2 — APPROVED** [review]

**Pull request merged**